labeller/multinamespacelistwatcher.go

```diff
@@ -0,0 +1,145 @@
+package labeller
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var resourceVersionSeparator = ","
+
+// Implements "k8s.io/client-go/tools/cache.ListerWatcher"
+type multiListerWatcher []cache.ListerWatcher
+
+func NewMultiNamespaceListerWatcher(getter cache.Getter, resource string, namespaces []string, fieldSelector fields.Selector) cache.ListerWatcher {
+	if len(namespaces) == 0 {
+		return cache.NewListWatchFromClient(getter, resource, corev1.NamespaceAll, fieldSelector)
+	}
+
+	if len(namespaces) == 1 {
+		return cache.NewListWatchFromClient(getter, resource, namespaces[0], fieldSelector)
+	}
+
+	var lws []cache.ListerWatcher
+	for _, ns := range namespaces {
```

Comment:
This puts more pressure on the kube API server, but it's our only option if we want to respect the RBAC permissions we are given.